File: src/main/java/com/kmarzecki/communicator/api/auth/AuthController.java
package com.kmarzecki.communicator.api.auth;

import com.kmarzecki.communicator.model.Language;
import com.kmarzecki.communicator.model.auth.LoginDto;
import com.kmarzecki.communicator.model.auth.LoginResponse;
import com.kmarzecki.communicator.model.auth.RegisterDto;
import com.kmarzecki.communicator.model.auth.UserResponse;
import com.kmarzecki.communicator.security.AuthenticationService;
import lombok.AllArgsConstructor;
import org.springframework.validation.annotation.Validated;
import org.springframework.web.bind.annotation.*;

import java.security.Principal;


/**
 *  Controller for operations concerning User registration and login processes
 */
@RestController
@RequestMapping("/api/auth")
@CrossOrigin(origins = {"*"},allowCredentials = "true", allowedHeaders = "*")
@AllArgsConstructor
public class AuthController {
    private final AuthenticationService authenticationService;

    /**
     * Login a user
     * @param request Request containing login information
     * @return logged user response
     */
    @PostMapping("/login")
    public LoginResponse login(
            @RequestBody @Validated LoginRequest request
    ) {
        return authenticationService.login(new LoginDto(request.getUsername(), request.getPassword()));
    }

    /**
     * Register a user
     * @param request Request containing registration information
     * @param language user language
     */
    @PostMapping("/register")
    public void register(@RequestBody RegisterRequest request) {
        authenticationService.register(new RegisterDto(request.getUsername(), request.getPassword(), request.getEmail()));
    }

    /**
     * Get information about the logged-in user
     * @param principal Principal of the asking user
     * @return Use information
     */
    @GetMapping("/me")
    public UserResponse getMe(Principal principal) {
        return authenticationService.getMe(principal);
    }
}

File: src/main/kotlin/com/kmarzecki/trollo/api/auth/AuthController.kt
package com.kmarzecki.trollo.api.auth

import com.kmarzecki.trollo.model.auth.LoginDto
import com.kmarzecki.trollo.model.auth.LoginResponse
import com.kmarzecki.trollo.model.auth.RegisterDto
import com.kmarzecki.trollo.model.auth.UserResponse
import com.kmarzecki.trollo.security.AuthenticationService
import org.springframework.validation.annotation.Validated
import org.springframework.web.bind.annotation.*
import java.security.Principal

/**
 * Controller for operations concerning User registration and login processes
 */
@RestController
@RequestMapping("/api/auth")
@CrossOrigin(origins = ["*"], allowCredentials = "true", allowedHeaders = ["*"])
class AuthController(
        val authenticationService: AuthenticationService
) {

    /**
     * Login a user
     * @param request Request containing login information
     * @return logged user response
     */
    @PostMapping("/login")
    fun login(
            @RequestBody @Validated request: LoginRequest
    ): LoginResponse? {
        return authenticationService.login(LoginDto(request.username, request.password))
    }

    /**
     * Register a user
     * @param request Request containing registration information
     */
    @PostMapping("/register")
    fun register(@RequestBody request: RegisterRequest) {
        authenticationService.register(RegisterDto(request.username, request.password, request.email))
    }

    /**
     * Get information about the logged-in user
     * @param principal Principal of the asking user
     * @return Use information
     */
    @GetMapping("/me")
    fun getMe(principal: Principal): UserResponse? {
        return authenticationService.getMe(principal)
    }
}